backend/src/main/java/com/htecgroup/skynest/util/JwtEmailVerificationUtils.java

```diff
@@ -0,0 +1,137 @@
+package com.htecgroup.skynest.util;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.JWTVerifier;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.exceptions.JWTVerificationException;
+import lombok.extern.log4j.Log4j2;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.util.Date;
+
+@Component
+@Log4j2
+public class JwtEmailVerificationUtils {
+  @Value("${backend.app.jwtEmailVerificationSecret}")
+  private String jwtEmailVerificationSecret;
+
+  @Value("${backend.app.jwtEmailVerificationExpirationMs}")
+  private int jwtEmailVerificationExpirationMs;
+
+  @Value("${backend.app.confirmationLink}")
+  private String confirmationLink;
+
+  public String generateJwtEmailVerificationToken(String email) {
+    return JWT.create()
+        .withSubject(email)
+        .withIssuedAt(new Date())
+        .withExpiresAt(new Date((new Date()).getTime() + jwtEmailVerificationExpirationMs))
+        .sign(Algorithm.HMAC256(jwtEmailVerificationSecret));
+  }
+
+  public String getEmailFromJwtEmailVerificationToken(String token) {
+    return JWT.require(Algorithm.HMAC256(jwtEmailVerificationSecret))
+        .build()
+        .verify(token)
+        .getSubject();
+  }
+
+  public boolean validateJwtToken(String token) {
+    try {
+      Date expiresAt =
+          JWT.require(Algorithm.HMAC256(jwtEmailVerificationSecret))
```

Comment:
You can extract this in variable: JWT.require(Algorithm.HMAC256(jwtEmailVerificationSecret)) and reuse it